### PR TITLE
implemented tests and algorithm

### PR DESCRIPTION
**Pull request by mayahp**
Created the meeting algorithm, made the given tests pass + wrote some more.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -28,247 +28,376 @@
...
-    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_60_MINUTES);
-
-    Collection<TimeRange> actual = query.query(events, request);
-    Collection<TimeRange> expected = Arrays.asList();
-
-    Assert.assertEquals(expected, actual);
-  }
-}
-
+    private static final Collection < Event > NO_EVENTS = Collections.emptySet();
```
Comment (danhoshow) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:31]: remove the unnecessary spaces... `private static final Collection<Event> NO_EVENTS = Collections.emptySet()`

Comment (danhoshow) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:31]: throughout the rest of this PR, too

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 2]
```diff
@@ -28,247 +28,376 @@
...
+
+    private FindMeetingQuery query;
+
+    @Before
+    public void setUp() {
+        query = new FindMeetingQuery();
+    }
+
+    @Test
+    public void optionsForNoAttendees() {
```
Comment (danhoshow) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:61]: In general (at Google), the names should "include the condition being tested and the expected outcome." I'm not sure if the test function names were given to you, but a name like `queryForNoAttendeesReturnsWholeDay` would be better

Comment (danhoshow) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:61]: if you could apply that throughout this file too

**Review by danhoshow — CHANGES_REQUESTED** [review]

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 2 above)
Comment (mayahp) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:61]: They gave us these test names, but I can edit the ones that I made up myself

**Review by danhoshow — APPROVED** [review]

**Pull request merged**